### PR TITLE
Ipce docks templates

### PR DESCRIPTION
**Pull request by SgStapleton**
This includes the new QComboBox implementation in ControlPointEditWidget, the new SaveAs option in TemplateEditorWidget, and the separation of map and registration template files within the project.

### CODE REVIEW

File: isis/src/qisis/objs/ControlPointEditWidget/ControlPointEditWidget.h [hunk 1]
```diff
@@ -231,7 +237,7 @@ namespace Isis {
       QPointer<QWidget> m_templateEditorWidget; //!< Template editor widget
       bool m_templateModified; //!< Indicates if the registration template was edited
 
-      QPointer<QLabel> m_templateFileNameLabel; //!< Label for the template filename
+      QPointer<QComboBox> m_templateComboBox; //!< ComboBox of importd registration templates
```
Comment (krlberry) [inline isis/src/qisis/objs/ControlPointEditWidget/ControlPointEditWidget.h]: Single character typo: Could you change `importd` to `imported` in the comment here?

File: isis/src/qisis/objs/ControlPointEditWidget/ControlPointEditWidget.cpp [hunk 2]
```diff
@@ -2386,6 +2409,59 @@ namespace Isis {
...
+  /**
+  * Appends the filename to the registrations path (unless this is the default template) and calls 
+  * setTemplateFile for the control measure
+  *
+  * @param filename This is the import directory and the filename of the template file selected from
+  *                 the QComboBox.
+  */
+  void ControlPointEditWidget::setTemplateFile(QString filename) {
+    QString expandedFileName = filename;
+    if(!filename.startsWith("$base")){
```
Comment (krlberry) [inline isis/src/qisis/objs/ControlPointEditWidget/ControlPointEditWidget.cpp:2436]: Will the template file ever be in a different area non-project area like `$hayabusa`?

File: isis/src/qisis/objs/ControlPointEditWidget/ControlPointEditWidget.cpp [hunk 3]
```diff
@@ -2386,6 +2409,59 @@ namespace Isis {
...
+  
+  
+  /**
+  * Reset the selected template in the template combobox if the template selected by the user does 
+  * not satisfy requirements for the control measure.
+  *
+  * @param fielName The filename that was previously selected in the template combo box
+  */
+  void ControlPointEditWidget::resetTemplateComboBox(QString fileName) {
+    if(fileName.startsWith("$base")) {
```
Comment (krlberry) [inline isis/src/qisis/objs/ControlPointEditWidget/ControlPointEditWidget.cpp]: (same comment as above)

File: isis/src/qisis/objs/ControlPointEditWidget/ControlPointEditWidget.cpp [hunk 4]
```diff
@@ -2386,6 +2409,59 @@ namespace Isis {
...
+      loadTemplateFile(expandedFileName);
+    }
+  }
+  
+  
+  /**
+  * Reset the selected template in the template combobox if the template selected by the user does 
+  * not satisfy requirements for the control measure.
+  *
+  * @param fielName The filename that was previously selected in the template combo box
```
Comment (krlberry) [inline isis/src/qisis/objs/ControlPointEditWidget/ControlPointEditWidget.cpp]: fielName -> fileName

File: isis/src/qisis/objs/Directory/Directory.cpp [hunk 5]
```diff
@@ -902,9 +907,6 @@ namespace Isis {
 
     connect( result, SIGNAL( destroyed(QObject *) ),
              this, SLOT( cleanupMatrixViewWidgets(QObject *) ) );
-
```
Comment (krlberry) [inline isis/src/qisis/objs/Directory/Directory.cpp]: Why were these removed? (I'm basically just double-checking to make sure that these were) deliberately removed. There's a lot of work in this PR and I couldn't figure out what other change caused this one.)

**Review by krlberry — COMMENTED** [review]
Looks good! A few very minor typos I found, and a question: 
For all of the lines in which there's something of the form 
value == "string" 
I just want to double-check on use of the case-sensitive test for equality -- will any of these values ever be capitalized (or in any way have a different case)?

File: isis/src/qisis/objs/Directory/ImportMapTemplateWorkOrder.cpp [hunk 6]
```diff
@@ -107,47 +107,25 @@ namespace Isis {
...
     QStringList templateFileNames;
 
     templateFileNames = QFileDialog::getOpenFileNames(
         qobject_cast<QWidget *>(parent()),
-        "Import " + itemType,
+        "Import Map Templates",
         QString(),
-        filterText,
-        &selectedFilter);
+        "Maps (*.map);; All Files (*)");
```
Comment (krlberry) [inline isis/src/qisis/objs/Directory/ImportMapTemplateWorkOrder.cpp:120]: Is it valid to have a mapfile with a filename like `map.pvl`? If not, this makes sense.

File: isis/src/qisis/objs/Directory/Directory.cpp (hunk 5 above)
Comment (SgStapleton) [reply isis/src/qisis/objs/Directory/Directory.cpp]: Yeah, sorry - this was an unrelated fix. These connections were not needed.

File: isis/src/qisis/objs/Directory/ImportMapTemplateWorkOrder.cpp (hunk 6 above)
Comment (SgStapleton) [reply isis/src/qisis/objs/Directory/ImportMapTemplateWorkOrder.cpp:120]: It is valid. I guess I am not sure of the concern here...? (If it is about the "Maps (*.map)..." line, this is simply a filter of the qfiledialog box. They can certainly get to .pvl files by choosing the "All Files" filter.)

File: isis/src/qisis/objs/ControlPointEditWidget/ControlPointEditWidget.cpp (hunk 2 above)
Comment (SgStapleton) [reply isis/src/qisis/objs/ControlPointEditWidget/ControlPointEditWidget.cpp:2436]: Not as of right now. The default template file in $base is hard-coded. The only other files that will exist as an option in this combobox are imported files. If other files are added in the future, this logic will need to change. Do you think this is reasonable now, or should I find another way of distinguishing?

**Comment by SgStapleton** [comment]
All instances of
value == "string"
currently will not have any other capitalization scheme and so do not need to be made case-insensitive.

File: isis/src/qisis/objs/ControlPointEditWidget/ControlPointEditWidget.cpp (hunk 2 above)
Comment (krlberry) [reply isis/src/qisis/objs/ControlPointEditWidget/ControlPointEditWidget.cpp:2436]: No, you don't need to change anything. If `$base` is hard coded for the template file area, the rest of this makes sense, and I think that it's okay that if this changes in the future, the other logic will need to be updated, too.

File: isis/src/qisis/objs/Directory/Directory.cpp (hunk 5 above)
Comment (krlberry) [reply isis/src/qisis/objs/Directory/Directory.cpp]: No problem! Just double-checking.

File: isis/src/qisis/objs/Directory/ImportMapTemplateWorkOrder.cpp (hunk 6 above)
Comment (krlberry) [reply isis/src/qisis/objs/Directory/ImportMapTemplateWorkOrder.cpp:120]: Then, no problem! I just wasn't picturing this correctly. Thanks!

**Comment by krlberry** [comment]
@SgStapleton :+1: Thanks! This looks good, so I'll go ahead and merge it.
